### PR TITLE
change Messenger API to accept PLUGIN_INFO object

### PR DESCRIPTION
**Pull request by joeyyang**
This PR makes the following changes:
- changes `new Messenger` API:
  - now accepts `pluginInfo` object, containing `name`, package `id`, and `majorVersion` number
  - signature is now `by: pluginInfo` instead of `by: string`
- updates Toggler to use the new `Messenger` API

This PR should be accompanied by corresponding PRs in all other plugins.

### CODE REVIEW

**Review by gblakeman — APPROVED** [review]
🛳💥

These changes are still working for me after some thought. 🙌🏼

**Review by lynnerang — APPROVED** [review]
LGTM!  Think we talked about this, but Specter shares the same PLUGIN_IDENTIFIER across all versions (1 & 2, betas for each), but looks like the name and majorVersion deets will be enough to differentiate between them all :)

**Comment by gblakeman** [comment]
@lynnerang Nice – @joeyyang can confirm, but I think we’re already tracking a `beta` flag, so between that and the version number we should be able to distinguish all the variations.

File: plugin-tools/src/Messenger.ts [hunk 1]
```diff
@@ -13,16 +13,16 @@ import { ApplicationInsights } from '@microsoft/applicationinsights-web';
  * Application Insights to use.
  */
 export default class Messenger {
-  name: string;
+  pluginInfo: any;
```
Comment (joeyyang) [inline plugin-tools/src/Messenger.ts:16]: TODO: Update all `new Messenger(...)` calls throughout plugin suite, and pair those PRs with this one.

File: toggler/src/GUI.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ const watchIncomingMessages = (): void => {
           updateSelected(isFrameSelected, userSelectedSource);
           break;
         case 'track':
-          trackEvent(PLUGIN_NAME, appInsightsKey, payload);
+          trackEvent(PLUGIN_INFO, appInsightsKey, payload);
```
Comment (joeyyang) [inline toggler/src/GUI.ts:86]: TODO: update `GUI.ts` throughout plugin suite to add `PLUGIN_INFO` object in `constants.ts` and pass that to `trackEvent` call.

**Pull request merged**